src/util/debug.c

```diff
@@ -287,7 +288,17 @@ void sss_vdebug_fn(const char *file,
          * searchable.
          */
         va_copy(ap_fallback, ap);
-        ret = journal_send(file, line, function, level, format, ap);
+        if (debug_chain_id > 0) {
+            ret = snprintf(new_format, sizeof(new_format),
+                           DEBUG_CHAIN_ID_FMT"%s", debug_chain_id, format);
```

Comment:
What happens if `snprintf` truncates here part of format, especially mid specifier..?

Reply:
Initially I meant to get rid of first allocation (`asprintf()`) but to keep allocation of `new_format`.

Getting rid of both mallocs would be even better, of course but this needs to be done safely.

One of the options, imo, could be:
 - make buffer slightly bigger (maybe 256 bytes?) so we can consider longer string to be exceptionally rare
 - check return of `snprintf()` vs `sizeof(new_format)` => if string didn't fit the buffer, fallback to malloc
This complicates code a bit, but should have a minimal performance effect.

Reply:
Changes made. Thank you for the suggestions, and the review.